app/src/main/java/com/codesoom/assignment/controllers/ProductController.java

```diff
@@ -1,42 +1,77 @@
 package com.codesoom.assignment.controllers;
 
+import com.codesoom.assignment.application.AuthenticationService;
 import com.codesoom.assignment.application.ProductService;
 import com.codesoom.assignment.domain.Product;
 import com.codesoom.assignment.dto.ProductData;
 import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.MissingRequestHeaderException;
 import org.springframework.web.bind.annotation.*;
 
 import javax.validation.Valid;
 import java.util.List;
 
+/**
+ * 상품에 대한 http request handler
+ * @see Product
+ * @see ProductService
+ */
```

Comment:
```suggestion
/**
 * 상품에 대한 http request handler
 */
```

두 `@see`는 불필요할 뿐 아니라 커플링을 발생시킵니다. 특히 `ProductService`를 javadoc에 명시하는 것은 `ProductController`가 `ProductService`를 사용한다는 사실을 공개하는 것과 비슷한 의미를 가져 바람직하지 않습니다. 클래스에 작성하는 javadoc은 클래스의 바깥이라 생각하고 작성해보세요.

Reply:
여기서 커플링의 의미가 무엇인가요? 그냥 사전적 의미는 연동된다 정도인데, 정확하게 안와닿아서요!

Reply:
의존관계가 생긴다는 뜻입니다. 커플링의 부작용 중 하나를 대표적으로 말하자면, 한 쪽의 수정사항이 다른 한 쪽에도 영향을 끼치게 되죠. 여기에서 만약 `ProductController`가 `ProductService`를 사용하지 않고 `ProductCommandService`를 사용하도록 수정되었다고 칩시다. 그렇다면 여기 주석까지 귀찮게 같이 수정해줘야 하는 문제가 생깁니다. "에이 주석인데 뭐 어때" 라는 생각을 할 수도 있을 거에요. 하지만 주석은 여기에서 겪을 수 있는 아주 단순한 예제일 뿐입니다. 많은 뛰어난 개발자들은 커플링에 대해 민감한 감각을 갖고 있습니다. 작은 것도 지나치지 않는 습관을 들여서 그런 감각을 키워나가 봅시다.

Reply:
@johngrib 오 친절한 설명감사합니다. 문득 궁금한 것이 그러한 커플링들을 시각화하거나 관리하게 해주는 툴 같은것은 없나요? 인간의 인지능력만으로 커버하기에는 에러가능성이 너무 클 것 같아서요

Reply:
인텔리J의 `Code > Analyze Code`에 관련 메뉴들이 있습니다. 그러나 모든 의존관계를 확실히 찾아주지는 못합니다. 도구의 사용법을 익히면서 좋은 습관을 기르고 감각을 함께 키우는 것이 가장 좋습니다.
 
![image](https://user-images.githubusercontent.com/1855714/133614197-4dc4aa4f-067c-42d5-9723-7d46174d87df.png)